connectors/src/connectors/notion/index.ts

```diff
@@ -217,3 +224,73 @@ export async function cleanupNotionConnector(
 
   return new Ok(undefined);
 }
+
+export async function retrieveNotionConnectorPermissions(
+  connectorId: ModelId,
+  parentInternalId: string | null
+): Promise<Result<ConnectorResource[], Error>> {
+  const c = await Connector.findOne({
+    where: {
+      id: connectorId,
+    },
+  });
+  if (!c) {
+    logger.error({ connectorId }, "Connector not found");
+    return new Err(new Error("Connector not found"));
+  }
+
+  const [pages, dbs] = await Promise.all([
+    NotionPage.findAll({
+      where: {
+        connectorId: connectorId,
+        parentId: parentInternalId || "workspace",
+      },
+    }),
+    NotionDatabase.findAll({
+      where: {
+        connectorId: connectorId,
+        parentId: parentInternalId || "workspace",
+      },
+    }),
+  ]);
+
+  const pageResources: ConnectorResource[] = await Promise.all(
+    pages.map((p) => {
+      return (async () => {
+        // Try to look for one NotionPage child to see if this page is expandable. Technically we
+        // should also look for NotionDatabase, but we don't support that to save one DB call here.
+        const child = await NotionPage.findOne({
+          where: {
+            connectorId: connectorId,
+            parentId: p.notionPageId,
+          },
+        });
+        const expandable = child ? true : false;
+
+        return {
+          provider: c.type,
+          internalId: p.notionPageId,
+          parentInternalId: null,
```

Comment:
Why do we set it to null here and below? When does this field is not null?

Reply:
Yeah that's an error will fix :+1: